runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1067,10 +1091,10 @@ NNFW_STATUS nnfw_session::set_config(const char *key, const char *value)
 
 const onert::ir::Graph *nnfw_session::primary_subgraph()
 {
-  if (_subgraphs)
+  if (_state == State::MODEL_LOADED)
   {
     assert(!_execution && _executions.empty());
-    return _subgraphs->primary().get();
+    return _model_graph->entry()->primary().get();
   }
   else
   {
```

Comment:
Assert below assumes if `_subgraphs` exists, there must be execution (either a single execution or executions (for experimental pipeline execution)). I don't understand why primary_subgraph() depends on internal state of [loaded, prepared, executed, ...]. It would be good to separate this concepts.